### ISSUE TITLE
vmwarevsphere issue uploading boot2docker.iso when Datacenter is under a folder

### ISSUE THREAD

**Issue by dannc** [docker/machine#3744]
I'm having an issue using the docker-machine vmwarevsphere driver to create machines on my vmware vcenter infrastructure.  Specifically I believe that it relates to having our DataCenter underneath a top-level folder in vmware.

I compiled the driver locally to help debug the issue.  Here's the output when trying to create the machine:

    ~/bin/docker-machine create --driver vmwarevsphere \
    --vmwarevsphere-vcenter lod-ftcolvc.lod.ca.lab \
    --vmwarevsphere-datastore Dynamic \
    --vmwarevsphere-username user@ca.com \
    --vmwarevsphere-password "this is not my password" \
    --vmwarevsphere-datacenter "Colorado Springs/Resource Pools" \
    --vmwarevsphere-hostsystem "Clustered Pools/foo.ca.com" \
    --vmwarevsphere-pool "/Colorado Springs/Resource Pools/host/Clustered Pools/Resources/OCTO" kubernetes1
    Running pre-create checks...
    Creating machine...
    (kubernetes1) Copying /Users/chuda04/.docker/machine/cache/boot2docker.iso to /Users/chuda04/.docker/machine/machines/kubernetes1/boot2docker.iso...
    (kubernetes1) Generating SSH Keypair...
    (kubernetes1) Creating VM...
    (kubernetes1) Uploading Boot2docker ISO ...
    (kubernetes1) Datastore URL: https://lod-ftcolvc.lod.ca.lab:443/folder/kubernetes1/boot2docker.iso?dcPath=Resource+Pools&dsName=Dynamic
    Error creating machine: Error in driver during machine creation: 404 Not Found

Since the DataCenter is under the "Colorado Springs" folder, the "dcPath" field should be set to "Colorado+Springs%2F/Resource+Pools" in the Datastore URL rather than just "Resource+Pools".  I hacked in the change to the URL on my local build to fixup the URL and I am able to create my docker-machine vm.

FWIW, it looks like the govmomi Datastore.URL function is the one that is returning the bare DataCenter name property in the URL but was hoping that maybe this could be fixed in the vmwarevsphere drivers since we already have access to the full DataCenter name (as supplied in this case through the --vmwarevsphere-datacenter option)

**Comment by brockz** [comment]
Can you show me the Code that you changed ?

**Comment by D3nn** [comment]
Is vsphere.go, i hacked around it like this:

	log.Infof("Uploading Boot2docker ISO ...")
	dsurl, err := dss.URL(ctx, dc, fmt.Sprintf("%s/%s", d.MachineName, isoFilename))
	log.Infof("Datastore URL: %s", dsurl)
	if err != nil {
		return err
	}
	dsurl, err = url.Parse(strings.Replace(dsurl.String(), "dcPath=", "dcPath=Colorado+Springs%2F", 1))
	if err != nil {
		return err
	}
	log.Infof("Datastore URL: %s", dsurl)

**Howerver**, you may not need to do this anymore because the underlying govmomi libraries were fixed.  See [Issue # 594](https://github.com/vmware/govmomi/issues/594) and [PR #635](https://github.com/vmware/govmomi/pull/635).  I haven't tested them yet but seems like they should fix the issue.

**Comment by brockz** [comment]
I have still the Problem with docker-machine v 0.9.0

**Comment by D3nn** [comment]
Yeah, I see no mention of them updating the vmware-vsphere driver in docker-machine 0.9.0.  Not sure when/how that gets done.

**Comment by kleinsasserm** [comment]
I have the same problem. in v. 0.12.0
Docker Machine Version:  0.12.0, build 45c69ad
Found binary path at /usr/local/bin/docker-machine
Launching plugin server for driver vmwarevsphere
Plugin server listening at address 127.0.0.1:45549
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(flag-lookup) Calling .GetMachineName
(flag-lookup) Calling .DriverName
(flag-lookup) Calling .GetCreateFlags
Found binary path at /usr/local/bin/docker-machine
Launching plugin server for driver vmwarevsphere
Plugin server listening at address 127.0.0.1:41885
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(test-docker) Calling .GetMachineName
(test-docker) Calling .DriverName
(test-docker) Calling .GetCreateFlags
(test-docker) Calling .SetConfigFromFlags
Running pre-create checks...
(test-docker) Calling .PreCreateCheck
(test-docker) DBG | Connecting to vSphere for pre-create checks...
(test-docker) Calling .GetConfigRaw
Creating machine...
(test-docker) Calling .Create
(test-docker) DBG | local Boot2Docker ISO version:  v17.05.0-ce
(test-docker) Copying /home/mario/.docker/machine/cache/boot2docker.iso to /home/mario/.docker/machine/machines/test-docker/boot2docker.iso...
(test-docker) Generating SSH Keypair...
(test-docker) Creating VM...
(test-docker) Uploading Boot2docker ISO ...
Error creating machine: Error in driver during machine creation: 404 Not Found
notifying bugsnag: [Error creating machine: Error in driver during machine creation: 404 Not Found]
